orchagent/cbf/nhgmaporch.cpp

```diff
@@ -0,0 +1,384 @@
+             * Extract the map from the values.
+             */
+            auto p = getMap(kfvFieldsValues(t));
+            success = p.first;
+
+            /*
+             * If the map can't be extracted, erase the work item as it's useless to retry unless the user updates the
+             * wrong values. We achieve the erase by setting the success value to "true".
+             *
+             */
+            if (!success)
+            {
+                SWSS_LOG_ERROR("Failed to extract NHG map %s", index.c_str());
+                success = true;
+            }
+            else
+            {
+                /*
+                * Create the SAI map.  Also track the largest index referenced by the map as we do.
+                */
+                auto *fc_map = new sai_map_t[p.second.size()];
+                uint32_t ii = 0;
+                int largest_nh_index = 0;
+
+                for (const auto &fc_nh_idx : p.second)
+                {
+                    fc_map[ii].key = fc_nh_idx.first;
+                    fc_map[ii++].value = fc_nh_idx.second;
+
+                    if (fc_nh_idx.second > largest_nh_index)
+                    {
+                        largest_nh_index = fc_nh_idx.second;
+                    }
+                }
+
+                sai_map_list_t fc_map_list;
+                assert(p.second.size() <= UINT32_MAX);
+                fc_map_list.count = static_cast<uint32_t>(p.second.size());
+                fc_map_list.list = fc_map;
+
+                /*
+                 * Create the map.
+                 */
+                if (fc_map_it == m_syncdMaps.end())
+                {
+                    /*
+                     * Check if we have enough resources for a new map.
+                     */
+                    if (m_syncdMaps.size() >= m_max_nhg_map_count)
+                    {
+                        SWSS_LOG_WARN("No more resources left for new next hop group map %s", index.c_str());
+                        success = false;
+                    }
+                    else
+                    {
+                        /*
+                        * Set the map attributes.
+                        */
+                        sai_attribute_t attr;
+                        vector<sai_attribute_t> attrs;
+
+                        attr.id = SAI_NEXT_HOP_GROUP_MAP_ATTR_TYPE;
+                        attr.value.u32 = SAI_NEXT_HOP_GROUP_MAP_TYPE_FORWARDING_CLASS_TO_INDEX;
+                        attrs.push_back(move(attr));
+
+                        attr.id = SAI_NEXT_HOP_GROUP_MAP_ATTR_MAP_TO_VALUE_LIST;
+                        attr.value.maplist = fc_map_list;
+                        attrs.push_back(move(attr));
+
+                        /*
+                        * Create the map over SAI.
+                        */
+                        sai_object_id_t nhg_map_id;
+                        sai_status_t status = sai_next_hop_group_api->create_next_hop_group_map(&nhg_map_id,
```

Comment:
Any CRM for next_hop_Group_map? If so, could you add it?

Reply:
There isn't a CRM for this

Reply:
I've now added a CRM